build: update to angular 19 and standalone components

LucaPezzato opened the pull request:
This PR contains:

- [ ] bugfix
- [x] feature
- [ ] refactor
- [ ] tests
- [ ] documentation
- [ ] others

### Breaking Changes?

- [ ] yes
- [x] no


### Describe Your Changes
Update the project dependencies and migrated all the components to standalone.

### Does this PR affects any existing issues?

- [x] yes
- [ ] no

[Feature request]: Angular 19 with Standalone components #591

sibiraj-s commented:
Hey @LucaPezzato thanks for the PR. its a long standing due on my side, couldn't find enough time to work on this. I'll be really helpful if you could update the docs as well, So it would be easy for me to review and get this out sooner. thanks. 

you can run the docs by 

```bash
cd docs/
npm run dev
```

LucaPezzato commented:
Hey @sibiraj-s, I've updated the documentation and renamed some components to include the Ngx prefix for better import clarity (`MenuComponent` was too generic)

LucaPezzato commented:
@sibiraj-s Any update on this ?

aurelien-altarriba commented:
I'm also waiting to be able to use the ngx editor in my angular 19 standalone, it's currently impossible to translate the interface since we can't use the forRoot.

sibiraj-s commented:
I apologize, I got super occupied on work. Will try to look and publish this weekend. I am keeping a remainder for this. Feel free to nudge in-case if I do.

The only part I need double check is the supplying configuration, other than that rest should be fairly straightforward.

sibiraj-s commented:
Thanks @LucaPezzato I'll try to release this soon. 

A general feedback, kindly avoid formatting changes while submitting any PR, since it makes the reviews harder and delays the merge. A PR always should only focus on what it is meant for.
